Commit ID: cacce25dbff6a5131bb4929047f4bf11cd06cf8b
Change ID: wmmtuktnyptuywlkmtxwuoprxpnolzku
Author   : flukey <flukey@vapourmail.eu> (2026-10-05 15:34:01)
Committer: flukey <flukey@vapourmail.eu> (2026-10-05 16:04:37)

    scripts gnu

diff --git a/scripts/fsadm.sh b/scripts/fsadm.sh
index 53afacca74..26239f29c3 100755
--- a/scripts/fsadm.sh
+++ b/scripts/fsadm.sh
@@ -261,7 +261,7 @@
 		MINOR=${MAJORMINOR##*:}
 		;;
 	  *)
-		STAT=$(stat --format "MAJOR=\$((0x%t)) MINOR=\$((0x%T))" "$RVOLUME")
+		STAT=$(stat -f "MAJOR=\$((0x%XHr)) MINOR=\$((0x%XLr))" "$RVOLUME")
 		test -n "$STAT" || error "Cannot get major:minor for \"$VOLUME\"."
 		eval "$STAT"
 		MAJORMINOR="${MAJOR}:${MINOR}"
@@ -316,7 +316,7 @@
 		read -r <"/sys/block/${VOL#/dev/}/dev" MOUNTEDMAJORMINOR 2>&1 || error "Cannot get major:minor for \"$VOLUME\"."
 		;;
 	  *)
-		STAT=$(stat --format "MOUNTEDMAJORMINOR=\$((0x%t)):\$((0x%T))" "$VOL")
+		STAT=$(stat -f "MOUNTEDMAJORMINOR=\$((0x%XHr)):\$((0x%XLr))" "$VOL")
 		test -n "$STAT" || error "Cannot get major:minor for \"$VOLUME\"."
 		eval "$STAT"
 		;;
@@ -393,7 +393,7 @@
 			MOUNTDEV=$(echo -n -e "${i%% *}")
 			MOUNTED=${i#* }
 			MOUNTED=$(echo -n -e "${MOUNTED%% *}")
-			STAT=$(stat --format "%d" "$MOUNTED")
+			STAT=$(stat -f "%d" "$MOUNTED")
 			validate_mounted_major_minor "$(decode_major_minor "$STAT")"
 		done < "$PROCMOUNTS"
 	fi
@@ -751,6 +751,7 @@
 ####################################
 diff_dates() {
 	echo "$(( $("$DATE" -u -d"$1" +%s 2>"$NULL") - $("$DATE" -u -d"$2" +%s 2>"$NULL") ))"
+         echo $(( $("$DATE" -j -f "%a %b %e %H:%M:%S %Y" "$1" +%s 2>"$NULL") - $("$DATE" -j -f "%a %b %e %H:%M:%S %Y" "$2" +%s 2>"$NULL") ))
 }
 
 check_luks() {
@@ -847,7 +848,7 @@
 TEST64BIT=$(( 1000 * 1000000000000 ))
 test "$TEST64BIT" -eq 1000000000000000 || error "Shell does not handle 64bit arithmetic."
 echo Y | "$GREP" Y >"$NULL" || error "Grep does not work properly."
-test "$("$DATE" -u -d"Jan 01 00:00:01 1970" +%s)" -eq 1 || error "Date translation does not work."
+test "$("$DATE" -j -f "%b %e %H:%M:%S %Y" "Jan 1 00:00:01 1970" +%s)" -eq 1 || error "Date translation does not work."
 
 
 if [ "$#" -eq 0 ]; then
diff --git a/scripts/lvm_import_vdo.sh b/scripts/lvm_import_vdo.sh
index d581f323db..16fee8225e 100755
--- a/scripts/lvm_import_vdo.sh
+++ b/scripts/lvm_import_vdo.sh
@@ -313,7 +313,7 @@
 		DEVMINOR=${MAJORMINOR##*:}
 		;;
 	  *)
-		RSTAT=$("$STAT" --format "DEVMAJOR=\$((0x%t)) DEVMINOR=\$((0x%T))" "$RDEVICE" || true)
+		RSTAT=$("$STAT" -f "DEVMAJOR=\$((0x%XHr)) DEVMINOR=\$((0x%XLr))" "$RDEVICE" || true)
 		[ -n "$RSTAT" ] || error "Cannot get major:minor for \"$DEVICE\"."
 		eval "$RSTAT"
 		;;
@@ -610,7 +610,7 @@
 	for i in $(awk '/.*device:/ {print $2}' "$TEMPDIR/vdoconf.yml"); do
 		local DEV
 		DEV=$("$READLINK" $READLINK_E "$i") || continue
-		RSTAT=$("$STAT" --format "MAJOR=\$((0x%t)) MINOR=\$((0x%T))" "$DEV" 2>/dev/null) || continue
+		RSTAT=$("$STAT" -f "MAJOR=\$((0x%XHr)) MINOR=\$((0x%XLr))" "$DEV" 2>/dev/null) || continue
 		eval "$RSTAT"
 		if [ "$MAJOR" = "$DEVMAJOR" ] && [ "$MINOR" = "$DEVMINOR" ]; then
 			[ -z "$FOUND" ] || error "VDO configuration contains duplicate entries $FOUND and $i."
@@ -725,4 +725,6 @@
 
 [ -n "$DEVICE" ] || error "Device name is not specified. (see: $TOOL --help)"
 
+"$READLINK" -e / >/dev/null 2>&1 || READLINK_E="-f"
+
 convert2lvm_
diff --git a/scripts/lvmdump.sh.in b/scripts/lvmdump.sh.in
index fbb5d2467c..3d0ed8bbb3 100644
--- a/scripts/lvmdump.sh.in
+++ b/scripts/lvmdump.sh.in
@@ -98,7 +98,7 @@
 	esac
 done
 
-NOW=$("$DATE" -u +%G%m%d%k%M%S | "$TR" -d ' ')
+NOW=$("$DATE" -u +%G%m%d%_H%M%S | "$TR" -d ' ')
 if test -n "$userdir"; then
 	dir=$userdir
 else
@@ -241,7 +241,7 @@
 
 myecho "Gathering /etc/lvm info..."
 log "$LS -laR /etc/lvm >> \"$dir/etc_lvm_listing\" 2>> \"$log\""
-log "$CP -RL --preserve=all /etc/lvm \"$dir/lvm\" 2>> \"$log\""
+log "$CP -RLp /etc/lvm \"$dir/lvm\" 2>> \"$log\""
 log "$LVM dumpconfig --type diff --file \"$dir/config_diff\" 2>> \"$log\""
 log "$LVM dumpconfig --type missing --file \"$dir/config_missing\" 2>> \"$log\""
 
@@ -328,8 +328,8 @@
 	log "$UDEVADM info --export-db >> \"$udev_dir/db\" 2>> \"$log\""
 	log "$CP -a /etc/udev/udev.conf \"$udev_dir/conf\" 2>> \"$log\""
 	log "$LS -la /lib/udev >> \"$udev_dir/lib_dir\" 2>> \"$log\""
-	log "$CP -RL --preserve=all /etc/udev/rules.d \"$udev_dir/rules_etc\" 2>> \"$log\""
-	log "$CP -RL --preserve=all /lib/udev/rules.d \"$udev_dir/rules_lib\" 2>> \"$log\""
+	log "$CP -RLp /etc/udev/rules.d \"$udev_dir/rules_etc\" 2>> \"$log\""
+	log "$CP -RLp /lib/udev/rules.d \"$udev_dir/rules_lib\" 2>> \"$log\""
 fi
 
 if (( lvmetad )); then
